<launch>
    <arg name="cam_cal_file" default="/home/omer/Desktop/deneme/TOF-VIO/config/flexx.yml"/>
    <node pkg="tof_vio" exec="nodelet_icp" name="vio_node" output="screen">

        <remap from="/input_gt"         to="/gt"/>
        <remap from="/input_imu"        to="/imu"/>
        <remap from="/input_tof_pc"     to="/points"/>
        <remap from="/input_tof_nir"    to="/image_nir"/>
        <remap from="/input_tof_depth"  to="/image_depth"/>
        <remap from="/NICP/out"         to="remapped_output"/>
        <!-- use other pipeline
        0:proposed method
        1:ICP+RANSAC
        2:Others
        -->
        <param name="icp/use_other_icp" value="0"/>
        <param name="icp/use_orig_pts"  value="false"/>
        <param name="icp/use_ransomdownsample_pts" value="false"/>
        <param name="icp/use_salient_pts" value="true"/>
        <param name="icp/use_robust_w" value="true"/>

        <!-- run in vo mode with out imu -->
        <param name="icp/vo_mode" value="false"/>
        <!-- Init Pose with Motion Capture system data -->
        <param name="icp/init_by_MC" value="true"/>
        <!-- Init Pose init at the center (0,0,0) with the IMU orientation-->
        <param name="icp/init_by_IMU" value="false"/>
        <!-- kf_criteria 0:use_1st_frame 1:frame count based 2:distance based-->
        <param name="icp/kf_criteria" value="2"/>

        <param name="icp/cam_cal_file" value="$(var cam_cal_file)"/>
        <param name="icp/sailentpts_sample_count" value="15000"/>
        <param name="icp/target_sailentpts_num" value="600"/>
        <param name="icp/use_depth_grad" value="1"/>
        <param name="icp/use_intensity_grad" value="1"/>
        <param name="icp/use_edge_detector" value="0"/>
        <param name="icp/use_canny" value="1"/>
        <param name="icp/sh_backfloor" value="0.01"/>
        <param name="icp/sh_depth" value="0.06"/>
        <param name="icp/sh_grey" value="70.0"/>
        <param name="icp/sh_edge" value="0.015"/>

    </node>

<!--################################################################################################-->
    <node pkg="tof_vio" exec="eskf" name="eskf" output="screen">
        <!-- <remap from="~imu"        to="/imu"/> -->
        <remap from="/vo"         to="/icp_odom"/>
        <param name="eskf/ng"           value="10.0"/>
        <param name="eskf/na"           value="10.0"/>
        <param name="eskf/nbg"          value="10.0"/>
        <param name="eskf/nba"          value="10.0"/>
        <param name="eskf/vo_p"         value="0.1"/>
        <param name="eskf/vo_q"         value="0.1"/>
        <param name="eskf/vo_delay_ms"  value="0.0"/>
   </node>
<!--################################################################################################-->
<!--    <node pkg="tof_vio" type="ekf" name="ekf" output="log">
        <remap from="~imu"        to="/imu"/>
        <remap from="~vo"         to="/gt/odom_gt"/>

        <param name="ekf/ng"      type="double"  value="50"/>
        <param name="ekf/na"      type="double"  value="50"/>
        <param name="ekf/nbg"     type="double" value="20"/>
        <param name="ekf/nba"     type="double" value="20"/>
        <param name="ekf/vo_p"    type="double" value="0.02"/>
        <param name="ekf/vo_q"    type="double" value="0.02"/>
    </node>-->
<!--################################################################################################-->
</launch>

